frontend/packages/pipelines-plugin/src/components/quicksearch/PipelineQuickSearchVersionDropdown.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { global_palette_green_500 as greenColor } from '@patternfly/react-tokens/dist/js/global_palette_green_500';
+import { CatalogItem } from 'packages/console-dynamic-plugin-sdk/src';
+import { isSelectedVersionInstalled } from './pipeline-quicksearch-utils';
+
+interface PipelineQuickSearchVersionDropdownProps {
+  selectedVersion: string;
+  item: CatalogItem;
+  onChange: (key: string) => void;
+}
+
+const PipelineQuickSearchVersionDropdown: React.FC<PipelineQuickSearchVersionDropdownProps> = ({
+  item,
+  onChange,
+  selectedVersion,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+  const toggleIsOpen = React.useCallback(() => setOpen((v) => !v), []);
+  const versions = item?.attributes?.versions ?? [];
+  const versionItems = versions.reduce((acc, { version, id }) => {
+    acc[id.toString()] = id === item.data?.latestVersion?.id ? `${version} (latest)` : version;
+    return acc;
+  }, {});
+
+  const getDropdownItem = React.useCallback(
+    (key: string): React.ReactNode => (
+      <DropdownItem
+        component="button"
+        key={key}
+        label={versionItems[key]}
+        onClick={(e) => {
+          e.stopPropagation();
+          onChange(key);
+          setOpen(false);
+        }}
+      >
+        <div className="opp-quick-search-details__version-dropdown-item">
+          {versionItems[key]}
+          {isSelectedVersionInstalled(item, key) && <CheckCircleIcon color={greenColor.value} />}
+        </div>
+      </DropdownItem>
+    ),
+    [item, onChange, versionItems],
+  );
+
+  return (
+    <Dropdown
+      data-test={'task-version'}
+      className="opp-quick-search-details__version-dropdown"
+      dropdownItems={Object.keys(versionItems).map(getDropdownItem)}
+      isOpen={isOpen}
+      toggle={
+        <DropdownToggle data-test={'task-version-toggle'} onToggle={toggleIsOpen}>
+          {versionItems[selectedVersion]}
```

Comment:
This can be undefined, no? Line 21 says:

```
  const versions = item?.attributes?.versions ?? [];
```

So when it's converted to an object, it could be missing the `selectedVersion` "id" -- this would make the toggle empty, no?

Reply:
I have a check in the caller, so it only renders this VersionDropdown component if its has some value.Now updated to return early if there are no versions available.